modules/library/jdbc/src/main/java/org/geotools/jdbc/InsertionClassifier.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.jdbc;
+
+import com.vividsolutions.jts.geom.Geometry;
+import org.geotools.factory.Hints;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.GeometryDescriptor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * PreparedStatement inserts must be split in function of those criteria:
+ * - useExisting
+ * - type of the geometries
```

Comment:
I don't think we really need to classify geometries based on their type, but only if they are null, or not. In most (all?) databases it's fine to have generic geometry columns that host everything, and the mechanisms we use to write them are not type specific to my knowledge, classifying by geometry type seems only a way to make batch inserts less effective.. unless I'm missing something?